src/creds/creds.c

```diff
@@ -181,8 +181,8 @@ static int add_credentials_to_table(Table *t, bool encrypted) {
                         if (r < 0)
                                 return log_error_errno(r, "Failed to determine backing file system of '%s': %m", de->d_name);
 
-                        secure = r ? "secure" : "weak"; /* ramfs is not swappable, hence "secure", everything else is "weak" */
-                        secure_color = r ? ansi_highlight_green() : ansi_highlight_yellow4();
+                        secure = r > 0 ? "secure" : "weak"; /* ramfs is not swappable, hence "secure", everything else is "weak" */
```

Comment:
so our coding style actually says that checking like this is ok for "boolean-like" values. This is kinda meant with that, i.e. stuff that is essentially a boolean, except that it can also carry an error code.

Of course, doing checks pretending it was a bool is only OK if we first checked that the value is not in that error code condition... which here took place above. (we often call these cases tri-state, for true/false/undefined)

So, dunno. I think we should accept both styles for cases like this, the explicit-numeric-comparison case as you switch this too here, but also the case teat-as-bool case.

or in other words, i think this commit is in line with coding style, but the status quo ante is too...

Reply:
Hmm, for most other functions where we return negative `errno`, `0`, or `1`, we still do the comparison explicitly. And this feels more intuitive for me.

Reply:
(i am fine merging your patch, if don't mind, just wanted to explain the situation)